TrainingApp/src/screens/common/product/product.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import { View, Image, Text, TouchableOpacity } from 'react-native';
+
+import styles, { ExtendedStyles } from './styles';
+
+class Product extends Component {
+    constructor(props) {
+        super(props);
+
+        this.styles = this.props.isExtended ? ExtendedStyles : styles;
+        this.currentImage = this.props.item.images[0];
+    }
+
+    getPriceString(item) {
+        return `$${item.price}`;
+    }
+
+    getOldPriceString(item) {
+        return `$${item.oldPrice}`;
+    }
+
+    getDiscountString(item) {
+        return `${item.discount}% Off`;
+    }
+
+    isActiveImage(image) {
+        return image.id === this.currentImage.id;
+    }
+
+    showImage(image) {
+        return (this.currentImage = image);
+    }
+
+    getControlsGSX() {
+        return this.props.isExtended ? (
+            <View style={this.styles.controlsWrapper}>
+                {this.props.item.images.map(image => (
+                    <View key={image.id}>{this.getControlGSX(image)}</View>
+                ))}
+            </View>
+        ) : null;
+    }
+
+    getStockButtonGSX() {
```

Comment:
change getStockButtonGSX to renderStockButtonGSX

Reply:
Renamed as mentioned